### PR TITLE
Skip PRs that don't actually merge

### LINKED ISSUE #100: Changelist gets created when a PR is opened
Hi, 

We observed that Changelists are getting added for a GUS work item when a PR is opened / re-opened. Please find that attached work item as a reference. 

https://gus.lightning.force.com/lightning/r/a07B0000008hw3cIAA/related/WorkChangeLists__r/view

I discussed this issue with Jag Jayaprakash and he suggested logging it here. Can someone please look into it as this causes confusion. 

We have seen this issue on multiple work items.

### PR DESCRIPTION
**Pull request by mpsenn**
Fixes #100

### CODE REVIEW

File: api/hooks/issues-hook/update-work-item-priority/__test__/updateWorkItemPriority.spec.js [hunk 1]
```diff
@@ -39,27 +39,27 @@ describe('updateWorkItemPriority issues hook', () => {
...
-        Issues.update.mockReset();
-        Issues.getByRelatedUrl.mockReset();
-        Issues.getByRelatedUrl.mockReturnValue(
-            Promise.resolve({
-                id: '1234abcd',
-                priority: 'P1'
-            })
-        );
+    describe('should not call Issues.update when the issue is linked but the priority is not great', () => {
+        // TODO set the 'priority' field to the right datatype once the todo in updateWorkItemPriority is resolved
```
Comment (AndrewHuffman) [inline api/hooks/issues-hook/update-work-item-priority/__test__/updateWorkItemPriority.spec.js:43]: Is this related to #100?

File: api/hooks/issues-hook/update-work-item-priority/index.js [hunk 2]
```diff
@@ -6,15 +6,23 @@
...
 const Issues = require('../../../services/Issues');
+const logger = require('../../../services/Logs/logger');
 
 module.exports = async function updateWorkItemPriority({
     priority,
     relatedUrl
 }) {
     const issue = await Issues.getByRelatedUrl(relatedUrl);
+
+    // TODO bug? is "priority" comparable with '>' ? The string "P2" is greater than "P1" but semantically its the other way around
```
Comment (AndrewHuffman) [inline api/hooks/issues-hook/update-work-item-priority/index.js:17]: Is this related to #100? Also, we should probably use issues instead of leaving TODO comments.

Comment (mpsenn) [reply api/hooks/issues-hook/update-work-item-priority/index.js:17]: Unrelated, this is a bug discovered that made me update the test above. Agreed, opened #150 for this.

File: api/actions/createChangelist.js [hunk 3]
```diff
@@ -24,6 +25,14 @@ module.exports = {
                 }
             }
         } = req.body;
+
+        if (merged_at === null) {
```
[the remark this answers is not in the document]
Comment (mpsenn) [reply api/actions/createChangelist.js]: The schema says string or null, but good point. Changed to `if (!merged_at) {` in latest.

File: package.json [hunk 4]
```diff
@@ -59,7 +59,8 @@
     "author": "reinierguerra",
     "license": "",
     "engines": {
-        "node": ">=10.15.1"
+        "node": "18.x",
+        "yarn": "1.22.19"
```
Comment (mpsenn) [inline package.json:63]: This is a *big* update. Will definitely need to test in staging before deploying this to Salesforce's prod app.

**Review by AndrewHuffman — APPROVED** [review]
lgtm, 🤞 the node update goes smoothly.

**Review by douglascayers — APPROVED** [review]
LGTM, thanks!
